R/predictiveAnalytics.R

```diff
@@ -73,6 +75,48 @@ predictiveAnalytics <- function(jaspResults, dataset, options) {
 
 }
 
+.predanErrorHandling <- function(dataset, options, ready){
+  if(!ready) return()
+
+  checks <- list(
+    .dateTimeArgsChecks <- function(){
+      timeVar <- try(as.POSIXct(dataset[[encodeColNames(options$time)]], tz = "UTC"))
+
+      if (isTryError(timeVar))
+        return(gettext("'Time' must be in a date-like format (e.g., yyyy-mm-dd hh:mm:ss or yyyy-mm-dd)"))
+
+    },
+    .predictionArgsChecks <- function(){
+      if(length(options$covariates) + length(options$factors) > 0 && options$trainingIndicator == "" &&  options$futurePredPredictionType != "noFuturePrediction")
+        return(gettext(paste("When 'Covariates' or 'Factors' are provided, they also need to be supplied for the future prediction period.",
+        "Please provide the 'Include in Training' variable where a value of '1' indicates that this period is used for training/verification - and a value of '0' that it is used for prediction.",
+        "The values for the dependent variable are allowed to be missing for the prediction period.",
+        "Alternatively, you could remove the covariates and select the specific time period you want to predict under 'Future Prediction' -> 'Periodical'.",
+        "That way the data is automatically extended into the future based on your settings. \n", 
+        "If you just want to check how well the predictions perform historically you can choose the option 'No forecast - verification only'")))
```

Comment:
you cannot use `paste` inside `gettext`, please format this as a single string. The same goes for the check below.